src/vm/assemblynative.cpp

```diff
@@ -1296,40 +1296,37 @@ INT_PTR QCALLTYPE AssemblyNative::GetLoadContextForAssembly(QCall::AssemblyHandl
    
     // Platform assemblies are semantically bound against the "Default" binder which could be the TPA Binder or
     // the overridden binder. In either case, the reference to the same will be returned when this QCall returns.
-    if (!pPEAssembly->IsProfileAssembly())
```

Comment:
If IsGAC wasn't lying, this should have been equivalent to `if (!false)`

If IsGAC was lying this seems like it still should be OK as `if (!AreSameBinderInstance(pTPABinder, pOpaqueBinder))` below should handle this being incorrect.